### ISSUE TITLE
rimraf node_modules delete all files

### ISSUE THREAD

**Issue by xiaoxue1341** [isaacs/rimraf#259]
I have a monorepo warehouse. When I delete node at the outermost layer_ Modules folder, the files in each folder under packages are also deleted
![image](https://user-images.githubusercontent.com/14305101/222639832-27ad6a88-51df-45a4-9cd6-86a56de430cd.png)

**Comment by isaacs** [comment]
That definitely shouldn't happen. What command are you running?

**Comment by edw1882** [comment]
Something like this I suspect 

`rimraf --glob node_modules */node_modules`

**Comment by isaacs** [comment]
Yeah, but have you tested that?  Because that is not what I'm seeing.

```bash
#!/bin/bash

rm -rf test-gh-259
mkdir test-gh-259
cd test-gh-259
mkdir -p node_modules/{a,b,c}
touch node_modules/{a,b,c}/{index.js,package.json}
mkdir -p packages/{x,y,z}/node_modules/{i,j,k}
touch packages/{x,y,z}/{node_modules/{i,j,k}/,}{index.js,package.json}
ln -s ../packages/x node_modules/x
ln -s ../packages/y node_modules/y
ln -s ../packages/z node_modules/z
ln -s ../x packages/y/node_modules/x
ln -s ../x packages/z/node_modules/x
ln -s ../y packages/x/node_modules/y
ln -s ../y packages/z/node_modules/y
ln -s ../z packages/x/node_modules/z
ln -s ../z packages/y/node_modules/z

tree
npx -y rimraf@latest --glob "node_modules" "packages/*/node_modules"
tree
```

Output:

```
.
├── node_modules
│   ├── a
│   │   ├── index.js
│   │   └── package.json
│   ├── b
│   │   ├── index.js
│   │   └── package.json
│   ├── c
│   │   ├── index.js
│   │   └── package.json
│   ├── x -> ../packages/x
│   ├── y -> ../packages/y
│   └── z -> ../packages/z
└── packages
    ├── x
    │   ├── index.js
    │   ├── node_modules
    │   │   ├── i
    │   │   │   ├── index.js
    │   │   │   └── package.json
    │   │   ├── j
    │   │   │   ├── index.js
    │   │   │   └── package.json
    │   │   ├── k
    │   │   │   ├── index.js
    │   │   │   └── package.json
    │   │   ├── y -> ../y
    │   │   └── z -> ../z
    │   └── package.json
    ├── y
    │   ├── index.js
    │   ├── node_modules
    │   │   ├── i
    │   │   │   ├── index.js
    │   │   │   └── package.json
    │   │   ├── j
    │   │   │   ├── index.js
    │   │   │   └── package.json
    │   │   ├── k
    │   │   │   ├── index.js
    │   │   │   └── package.json
    │   │   ├── x -> ../x
    │   │   └── z -> ../z
    │   └── package.json
    └── z
        ├── index.js
        ├── node_modules
        │   ├── i
        │   │   ├── index.js
        │   │   └── package.json
        │   ├── j
        │   │   ├── index.js
        │   │   └── package.json
        │   ├── k
        │   │   ├── index.js
        │   │   └── package.json
        │   ├── x -> ../x
        │   └── y -> ../y
        └── package.json

23 directories, 36 files
.
└── packages
    ├── x
    │   ├── index.js
    │   └── package.json
    ├── y
    │   ├── index.js
    │   └── package.json
    └── z
        ├── index.js
        └── package.json

4 directories, 6 files
```

**Comment by isaacs** [comment]
It's just deleting the node_modules folders, not the package contents themselves.

**Comment by isaacs** [comment]
Replacing the rimraf command with `npx -y rimraf@latest --glob '**/node_modules'` has the same effect.  Removes node_modules, not the package source contents.

**Comment by isaacs** [comment]
Will re-open if reproduction case is provided.

**Comment by xiaoxue1341** [comment]
> Will re-open if reproduction case is provided.


https://user-images.githubusercontent.com/14305101/223045006-3c1ddf1c-2f39-4d7e-b833-5f148c232b2c.mp4

You can see this video. All files in liveRoom2 have been deleted. The file on the right shows 3k+deleted files

**Comment by xiaoxue1341** [comment]
https://user-images.githubusercontent.com/14305101/223045166-dd8f4bea-d0c4-4376-a02d-5e0772739cbb.mp4

You can see this video. All the files in liveRoom2 have been deleted. The file change area on the left shows the deleted 3k+files

**Comment by isaacs** [comment]
The video is not loading. Please share what command you ran. Just comment as text, not a video or screenshot.

**Comment by edw1882** [comment]
He's on windows and running

rimraf node_modules

It is deleting ALL the files in a folder further down the tree that happens to have a node_modules folder.

**Comment by isaacs** [comment]
Ah, I think I see what's going on. Symlinks aren't being handled properly in the windows version. Will fix shortly.

**Comment by isaacs** [comment]
Fixed on 4.3.1. It could also have affected posix systems, so this is potentially serious.  Thanks for reporting it.
